diff --git a/BUILD b/BUILD
index b841f33..2841fe4 100644
--- a/BUILD
+++ b/BUILD
@@ -1,69 +1,68 @@
-load("@com_github_grpc_grpc//bazel:cc_grpc_library.bzl", "cc_grpc_library")
+load("@rules_proto//proto:defs.bzl", "proto_library")
+load("@rules_proto_grpc_cpp//:defs.bzl", "cc_proto_library", "cc_grpc_library")
+
+package(default_visibility = ["//visibility:public"])
 
 cc_grpc_library(
     name = "attention_broker_cc_grpc",
-    srcs = [":attention_broker_proto"],
+    protos = [":attention_broker_proto"],
     grpc_only = True,
-    visibility = ["//visibility:public"],
     deps = [":attention_broker_cc_proto"],
 )
 
 cc_grpc_library(
     name = "distributed_algorithm_node_cc_grpc",
-    srcs = [":distributed_algorithm_node_proto"],
+    protos = [":distributed_algorithm_node_proto"],
     grpc_only = True,
-    visibility = ["//visibility:public"],
     deps = [":distributed_algorithm_node_cc_proto"],
 )
 
 cc_grpc_library(
     name = "common_cc_grpc",
-    srcs = [":common_proto"],
+    protos = [":common_proto"],
     grpc_only = True,
-    visibility = ["//visibility:public"],
     deps = [":common_cc_proto"],
 )
 
 cc_proto_library(
     name = "attention_broker_cc_proto",
-    visibility = ["//visibility:public"],
-    deps = [":attention_broker_proto"],
+    protos = [":attention_broker_proto"],
+    deps = [
+        ":common_cc_proto"
+    ]
 )
 
 cc_proto_library(
     name = "distributed_algorithm_node_cc_proto",
-    visibility = ["//visibility:public"],
-    deps = [":distributed_algorithm_node_proto"],
+    protos = [":distributed_algorithm_node_proto"],
+    deps = [
+        ":common_cc_proto"
+    ]
 )
 
 cc_proto_library(
     name = "common_cc_proto",
-    visibility = ["//visibility:public"],
-    deps = [":common_proto"],
+    protos = [":common_proto"],
 )
 
 proto_library(
     name = "attention_broker_proto",
     srcs = ["attention_broker.proto"],
-    visibility = ["//visibility:public"],
     deps = [":common_proto"],
 )
 
 proto_library(
     name = "distributed_algorithm_node_proto",
     srcs = ["distributed_algorithm_node.proto"],
-    visibility = ["//visibility:public"],
     deps = [":common_proto"],
 )
 
 proto_library(
     name = "common_proto",
     srcs = ["common.proto"],
-    visibility = ["//visibility:public"],
 )
 
 proto_library(
     name = "echo_proto",
     srcs = ["echo.proto"],
-    visibility = ["//visibility:public"],
 )
diff --git a/MODULE.bazel b/MODULE.bazel
index bd75b85..1f29c15 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -17,5 +17,5 @@ bazel_dep(name = "rules_go", version = "0.52.0", repo_name = "io_bazel_rules_go"
 bazel_dep(name = "googleapis", version = "0.0.0-20240819-fe8ba054a", repo_name = "com_google_googleapis")
 bazel_dep(name = "gazelle", version = "0.41.0", repo_name = "bazel_gazelle")
 bazel_dep(name = "protobuf", version = "29.3", repo_name = "com_google_protobuf")
-bazel_dep(name = "grpc", version = "1.69.0", repo_name = "com_github_grpc_grpc")
+bazel_dep(name = "grpc", version = "1.76.0.bcr.1", repo_name = "com_github_grpc_grpc")
 
